### PR TITLE
GH2063: Adds netstandard 2 to script conventions

### LINKED ISSUE #2063: Cake running on Mono can't load netstandard 2.0 assembly
Currently getting `You must add a reference to assembly 'netstandard, Version=2.0.0.0`

Minimal repro:
```
#addin nuget:?package=Microsoft.AspNetCore.TestHost&version=2.0.0&loaddependencies=true
#addin nuget:?package=Microsoft.AspNetCore.Http.Extensions&version=2.0.0&loaddependencies=true

using Microsoft.AspNetCore.Builder;
using Microsoft.Extensions.Configuration;

Information("Oh hi, Mark!");

new ConfigurationBuilder().Build();
```

Gives:
```
standard.cake(9,28): error CS0012: The type 'Object' is defined in an assembly that is not referenced. You must add a reference to assembly 'netstandard, Version=2.0.0.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51'.
```

### PR DESCRIPTION
**Pull request by mholo65**
- Fixes #2063

### CODE REVIEW

**Comment by mholo65** [comment]
@cake-build/cake-team if we'd like to create a hotfix for this, we should re-target this PR.

File: src/Cake.Core/Scripting/ScriptConventions.cs [hunk 1]
```diff
@@ -85,6 +85,7 @@ public IReadOnlyList<Assembly> GetDefaultAssemblies(DirectoryPath root)
             // This is just to please Roslyn when running under Mono. See issue https://github.com/dotnet/roslyn/issues/19364
             result.Add(_loader.Load(new AssemblyName("System.Runtime, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a"))); // System.Runtime
             result.Add(_loader.Load(new AssemblyName("System.Collections, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a"))); // System.Collections
+            result.Add(_loader.Load(new AssemblyName("netstandard, Version=2.0.0.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51"))); // NetStandard.Library
```
Comment (devlead) [inline src/Cake.Core/Scripting/ScriptConventions.cs]: Add a try/catch diagnostic log here.

Confirmed working on mono 5.4 but fails on older versions.

**Comment by gep13** [comment]
@mholo65 I have re-targetted this PR against a new `hotfix/0.26.1` branch.

Let me know if you have any questions.

**Comment by mholo65** [comment]
@gep13 Thanks! I've set correct milestone to the related issue.

File: src/Cake.Core/Scripting/ScriptConventions.cs [hunk 2]
```diff
@@ -85,6 +85,19 @@ public IReadOnlyList<Assembly> GetDefaultAssemblies(DirectoryPath root)
...
+            try
+            {
+                result.Add(_loader.Load(new AssemblyName("netstandard, Version=2.0.0.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51"))); // NETStandard.Library
+            }
+            catch
+            {
+                /*
+                 * Silently continue instead and let it blow up during runtime if netstandard assembly was not found.
+                 * TODO: Log that netstandard assembly was not found.
+                 * Unfortunately, logger is not available in this class, and that would be too big of a change for the 0.26.1 hotfix release.
```
Comment (patriksvensson) [inline src/Cake.Core/Scripting/ScriptConventions.cs]: Is it really that big change to inject the logger and write the message here?

Comment (mholo65) [reply src/Cake.Core/Scripting/ScriptConventions.cs]: Well, class is public and constructor also. A new `internal` constructor however could be done. As this is for a hotfix, only internal changes are allowed.

Comment (patriksvensson) [reply src/Cake.Core/Scripting/ScriptConventions.cs]: Ah, ok.

**Comment by devlead** [comment]
LGMT 👍

**Pull request merged**